src/protocolsupport/protocol/typeremapper/pe/inventory/PESlotRemapper.java

```diff
@@ -1,13 +1,270 @@
 package protocolsupport.protocol.typeremapper.pe.inventory;
 
+import protocolsupport.api.ProtocolVersion;
+import protocolsupport.protocol.packet.middleimpl.ClientBoundPacketData;
+import protocolsupport.protocol.packet.middleimpl.clientbound.play.v_pe.InventorySetItems;
+import protocolsupport.protocol.packet.middleimpl.clientbound.play.v_pe.InventorySetSlot;
 import protocolsupport.protocol.packet.middleimpl.serverbound.play.v_pe.GodPacket.InvTransaction;
 import protocolsupport.protocol.storage.netcache.NetworkDataCache;
+import protocolsupport.protocol.storage.netcache.PEInventoryCache;
 import protocolsupport.protocol.storage.netcache.WindowCache;
 import protocolsupport.protocol.typeremapper.pe.inventory.PEInventory.PESource;
 import protocolsupport.protocol.utils.types.networkentity.NetworkEntity;
+import protocolsupport.utils.recyclable.RecyclableArrayList;
+import protocolsupport.zplatform.itemstack.ItemStackWrapper;
 
 public class PESlotRemapper {
-	
+
+	public static ClientBoundPacketData remapClientBoundSlot(NetworkDataCache cache, ProtocolVersion version, ItemStackWrapper item, int slot) {
+		String locale = cache.getAttributesCache().getLocale();
+		PEInventoryCache invCache = cache.getPEInventoryCache();
+		WindowCache winCache = cache.getWindowCache();
+		switch(cache.getWindowCache().getOpenedWindow()) {
```

Comment:
omg big switcher. use fastutil and maps..

Reply:
Maybe... But the remapping is all very dependend on the type of inventory opened. I could remap with windowId and slot into one integer but then we have a giant map and using that with the inventorySetItems will actually cause way worse performance.

Reply:
Okey.